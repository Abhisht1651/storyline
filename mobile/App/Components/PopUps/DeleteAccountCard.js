// @flow
import React from 'react'
import { View, Text, TouchableOpacity, Image } from 'react-native'
import Images from '../../Themes/Images'
import styles from '../Styles/PopUpStyles/DeleteAccountCardStyle'

type Props = {
  onBtnPress: () => void,
  onClose: () => void
}

const DeleteAccountCard = (props: Props) => {
  const { onBtnPress, onClose } = props
  return (
    <View style={styles.card}>
      <TouchableOpacity style={styles.closeIconContainer} activeOpacity={1} onPress={onClose}>
        <Image source={Images.closeIconPurple} style={styles.closeIcon} />
      </TouchableOpacity>
      <View style={styles.textContainer}>
        <Text style={styles.deleteAccountText}>Delete account</Text>
        <Text style={styles.infoText}>
          Are you sure you want to delete this account? You can never recover your account in
          future.
        </Text>
      </View>
      <TouchableOpacity
        style={styles.continueButtonContainer}
        activeOpacity={1}
        onPress={onBtnPress}>
        <Text style={styles.continueButtonText}>Continue</Text>
      </TouchableOpacity>
    </View>
  )
}

export default DeleteAccountCard
