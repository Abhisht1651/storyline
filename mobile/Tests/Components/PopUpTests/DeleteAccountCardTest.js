import { shallow } from 'enzyme'
import React from 'react'

import DeleteAccountCard from '../../../App/Components/PopUps/DeleteAccountCard'

describe('DeleteAccountCard', () => {
  describe('Rendering', () => {
    it('should match to snapshot', () => {
      const component = shallow(<DeleteAccountCard />)
      expect(component).toMatchSnapshot()
    })
  })
})

it('should render the Text Component', () => {
  const wrapper = shallow(<DeleteAccountCard />)
  expect(wrapper.find('Text')).toHaveLength(3)
})
