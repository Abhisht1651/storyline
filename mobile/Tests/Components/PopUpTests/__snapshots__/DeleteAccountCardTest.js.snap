// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeleteAccountCard Rendering should match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderRadius": 15,
      "height": 486.2857142857143,
      "justifyContent": "center",
      "width": 650,
    }
  }
>
  <ForwardRef
    activeOpacity={1}
    style={
      Object {
        "alignItems": "flex-end",
        "height": 31.21428571428571,
        "justifyContent": "flex-end",
        "width": "84%",
      }
    }
  >
    <Image
      source={
        Object {
          "default": "",
        }
      }
      style={
        Object {
          "aspectRatio": 1,
          "height": 31.21428571428571,
          "tintColor": "#1E1E60",
        }
      }
    />
  </ForwardRef>
  <View
    style={
      Object {
        "alignItems": "center",
        "height": "51%",
        "justifyContent": "center",
        "marginVertical": 8.214285714285714,
        "width": "80%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1E1E60",
          "fontFamily": "BananaGrotesk-Bold",
          "fontSize": 26,
          "letterSpacing": -0.14,
          "marginBottom": 18.07142857142857,
        }
      }
    >
      Delete account
    </Text>
    <Text
      style={
        Object {
          "color": "#1F1F60",
          "fontFamily": "BananaGrotesk-Light",
          "fontSize": 20.8,
          "letterSpacing": -0.11,
          "lineHeight": 41.07142857142857,
          "textAlign": "center",
        }
      }
    >
      Are you sure you want to delete this account? You can never recover your account in future.
    </Text>
  </View>
  <ForwardRef
    activeOpacity={1}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#7B46E4",
        "borderRadius": 39,
        "height": "18%",
        "justifyContent": "center",
        "marginVertical": 8.214285714285714,
        "width": "84%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontFamily": "BananaGrotesk-Bold",
          "fontSize": 20.8,
          "letterSpacing": -0.09,
        }
      }
    >
      Continue
    </Text>
  </ForwardRef>
</View>
`;
